endpoints/openrtb2/auction.go

```diff
@@ -1946,3 +1912,18 @@ func checkIfAppRequest(request []byte) (bool, error) {
 	}
 	return false, nil
 }
+
+func validateStoredBidRespAndImpExtBidders(bidderExts map[string]json.RawMessage, storedBidResp stored_responses.ImpBidderStoredResp, impId string) []error {
+	if bidResponses, ok := storedBidResp[impId]; ok {
+		storedBidRespValidationError := []error{fmt.Errorf("request validation failed. Stored bid responses are specified for imp %s. Bidders specified in imp.ext should match with bidders specified in imp.ext.prebid.storedbidresponse", impId)}
+		if len(bidResponses) < len(bidderExts)-1 { // do not count "prebid" part of imp.ext. It is always present in imp.ext if stored bid responses are specified. There are can be other properties except of bidders and "prebid"
+			return storedBidRespValidationError
+		}
```

Comment:
It seems like there is still a problem of some kind here. Perhaps this `if` block should be removed because I believe the following is a valid case that should result in both `appnexus` and `openx` http requests with the `appnexus` one being a real request sent to the bidder and `openx` one tied to a stored bid response. Right now I'm getting a validation error on this line for this case.
```
"ext": {
    "appnexus": {
        "placementId": 14997137
    },
    "openx": {
        "unit": "123",
        "platform": "ec25c60e-00f3-4861-a5ec-65c98349a153"
    },
    "prebid": {
        "storedbidresponse": [
            {
                "bidder": "appnexus",
                "id": "d3fa8c49-c36a-e2a5-a1fe-1f90afa27b93"
            }
        ]
    }
}
```
I ran this test case against PBS-Java and this does result in both `appnexus` and `openx` showing up in `httpCalls`.

Given the above test scenario, I removed the `if` statement, and it appears that only an `appnexus` bid request is being created since I only see `appnexus` and not `openx` in `httpCalls`.

Reply:
Discussed offline. PBS-Go has made a conscious decision to not send any real bidder requests if `storedbidresponse` is present in the request which is why `openx` does not appear in `httpCalls` for the example above.